### PR TITLE
prevent deleting subobjects

### PR DESCRIPTION
**Pull request by alonfirestein**

### CODE REVIEW

File: controllers/flashsystemcluster_controller.go [hunk 1]
```diff
@@ -416,15 +417,22 @@ func (r *FlashSystemClusterReconciler) ensureExporterDeployment(instance *odfv1a
 		r.Log.Error(err, "failed to create exporter deployment")
 		return err
 	}
-
+	newOwnerDetails := v1.OwnerReference{
```
Comment (bvered) [inline controllers/flashsystemcluster_controller.go:420]: you can create newOwnerDetails once in the reconcile method and send it as a parameter to all sub-method below like ensureExporterDeployment,  ensureExporterService ... 

that way you won't duplicate code in all 3 methods

Comment (alonfirestein) [reply controllers/flashsystemcluster_controller.go]: Fixed

File: controllers/flashsystemcluster_controller.go [hunk 2]
```diff
@@ -479,14 +494,21 @@ func (r *FlashSystemClusterReconciler) ensureExporterServiceMonitor(instance *od
 		r.Log.Error(err, "failed to get exporter servicemonitor")
 		return err
 	}
+	if reflect.DeepEqual(foundServiceMonitor.Spec, expectedServiceMonitor.Spec) {
+		r.Log.Info("existing exporter deployment is expected with no change")
+		foundServiceMonitor.SetOwnerReferences(append(foundServiceMonitor.GetOwnerReferences(), newOwnerDetails))
```
Comment (tyichye19) [inline controllers/flashsystemcluster_controller.go:499]: This line has no effect, as there is no call to update the object on the cluster (`r.Client.Update`)

**Comment by alonfirestein** [comment]
[Multi storage- on storageSystem deletion action - don't delete sub-objects (deployment / service / pod /repilca..) unless it's the last storageSystem object](https://trello.com/c/Am9R3QAe/92-multi-storage-on-storagesystem-deletion-action-dont-delete-sub-objects-deployment-service-pod-repilca-unless-its-the-last-storag)

**Review by tyichye19 — APPROVED** [review]
LGTM

**Review by bvered — APPROVED** [review]

**Pull request merged**